hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/HiveSchemaUtil.java

```diff
@@ -327,20 +284,14 @@ private static String createHiveMap(String keyType, String valueType) {
   /**
    * Create an Array Hive schema from equivalent parquet list type.
    */
-  private static String createHiveArray(Type elementType, String elementName) {
+  private static String createHiveArrayFromAvro(String schemaName, Schema elementType) {
     StringBuilder array = new StringBuilder();
     array.append("ARRAY< ");
-    if (elementType.isPrimitive()) {
-      array.append(convertField(elementType));
+    if (elementType.getType().equals(Schema.Type.RECORD) && (elementType.getFields().size() == 1
+        && !elementType.getName().equals("array") && !elementType.getName().endsWith("_tuple"))) {
```

Comment:
Wondering if this conditions is really needed ?

Reply:
I am not very conversant with these conversions. But if we can ensure we have full coverage of tests for all scenarios, should be easy to validate if these are required.